src/main/java/seedu/address/model/activity/Activity.java

```diff
@@ -64,6 +64,24 @@ public Activity(Title title, Integer ... ids) {
         this(primaryKeyCounter++, title, ids);
     }
 
+    /**
+     * Returns a new Activity instance with fields modified from the supplied activity.
+     * @param activity Activity instance to duplicate.
+     * @param title Title of the updated activity.
+     * @return a new Activity instance with the editable fields updated.
+     */
+    public Activity(Activity activity, Title title) {
```

Comment:
code duplication can be reduced with `this()`

Reply:
But need to keep all the expenses and existing debt info though o_o wouldn't `this()` reset the info

Reply:
I'm not sure but I think this is effectively a copy constructor - I'm okay with this right now since we have no other need to copy an `Activity` presently.